pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +112,313 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil // success
+}
+
+func convertOvnEgressIpToSdnEgressIp(ctx context.Context, client cnoclient.Client) error {
+	egressIpRollbackReady, err := sdnEgressIpResourcesReady(ctx, client)
+	if !egressIpRollbackReady {
+		return nil // wait for all SDN egressIP resources to be created before rollback
+	} else if err != nil {
+		return err
+	}
+
+	// 1. query for egressips
+	eipList, err := client.Default().Dynamic().Resource(gvrEgressIp).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// 2. query for nodes
+	nodeList, err := client.Default().Kubernetes().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// 3. iterate through egressIP objects and generate netnamespaces
+	//    - value of `namespaceSelector` will determine netnamespace's name
+	//    - egressIP list will be copied to egressIPs field for netnamespace
+	for _, eip := range eipList.Items {
+		// translate egressIPs to SDN netnamespaces
+		// - delete cloudprivateipconfig if it exists
+		// - create egressip to netnamespace resources for sdn
+
+		egressIps := eip.Object["spec"].(map[string]interface{})["egressIPs"].([]interface{})
+
+		for _, egressIp := range egressIps {
+			egressIpStr := fmt.Sprintf("%v", egressIp)
+			err = client.Default().Dynamic().Resource(gvrCloudPrivateIPConfig).Delete(ctx, egressIpStr, metav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+		}
+
+		eipNamespace := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+		eipNamespaceStr := fmt.Sprintf("%v", eipNamespace)
+		nns, err := client.Default().Dynamic().Resource(gvrNetnamespace).Get(ctx, eipNamespaceStr, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		nnsResourceVersion := nns.Object["metadata"].(map[string]interface{})["resourceVersion"]
+		nnsNetId := nns.Object["netid"]
+		netName := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+
+		netnamespace := &uns.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "network.openshift.io/v1",
+				"kind":       "NetNamespace",
+				"metadata": map[string]interface{}{
+					"name":            netName,
+					"resourceVersion": nnsResourceVersion,
+				},
+				"netid":     nnsNetId,
+				"netname":   netName,
+				"egressIPs": egressIps,
+			},
+		}
+		_, err = client.Default().Dynamic().Resource(gvrNetnamespace).Update(ctx, netnamespace, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	// 4. iterate through nodes and generate hostsubnets for any that are egress-assignable
+	for _, node := range nodeList.Items {
+		if _, ok := node.Labels["k8s.ovn.org/egress-assignable"]; ok {
+			// parse out node's subnet from egressip annotation
```

Comment:
I think we need to document this. It means users cannot expect the hostsubnet object will be exactly the same as the one before migration.

Reply:
If the SDN use case is normal and they have hostsubnet object using node subnet CIDR for `EgressCIDRs`, the object will be the exact same as before migration. However if they use some other value for `EgressCIDRs` the configuration rollback migration will generate will not be exactly the same, right, I agree. Definitely worth documenting these details!
There isn't a way to rollback perfectly for every possible use case the SDN API allows.

Reply:
Can we store this information in annotations on the resources created on migration and use it for the rollback?

Reply:
For the manual case, users should not expect to get this information back on rollback, because support for manual migration is very limited. 

For automatic case, we can store this information somewhere and reuse it for rollback, sure. I am thinking it is acceptable to preserve the EgressCIDRs field by labeling the node with it. On rollback we can read this value if it exists. 

To avoid error in a manual config scenario, if the value does not exist, we default to node subnet.

Reply:
Yeah, I was just stating the idea, did not actually expect to be done in this PR.

Reply:
(Leaving this unresolved so we don't miss the note on documentation)

Reply:
Actually, I'm going to resolve this. Because I have since added the feature Jaime suggested for automatic config, there is not much nuance here. 

The config will be restored in the automatic case, so there is nothing worth documenting because behavior is as expected. 

For manual case we don't support officially in the first place, so we don't need to provide more details for a "use at your own risk" feature.